includes/subscriptions/class-wc-payments-product-service.php

```diff
@@ -128,11 +128,12 @@ public function get_wcpay_product_id( WC_Product $product, $test_mode = null ) :
 	 * return string       The item's WCPay product id.
 	 */
 	public function get_wcpay_product_id_for_item( string $type ) : string {
-		if ( ! get_option( self::get_wcpay_product_id_option() . '_' . $type ) ) {
-			$this->create_product_for_item_type( $type );
+		$sanitized_type = str_replace( ' ', '_', trim( strtolower( $type ) ) );
```

Comment:
This will effectively remove all spaces and replace them with `_`, I wonder if that's the only thing we want to do? ie, are there any other things we want to sanitize? Maybe `+`, `%` or other special characters...

Also, can we re-use some of the existing Woo functionality? The `sanitize_key()` function for example.

Reply:
A great start, @nagpai! :)

I was having thoughts similar to @elazzabi just now after looking at it more carefully. `sanitize_key` is purpose-built for scenarios like this and ensures no extraneous characters are present in the target string.

Going by its [source code](https://core.trac.wordpress.org/browser/tags/6.0/src/wp-includes/formatting.php#L2154), looks like it will remove even spaces. Although that might result in reduced readability compared to what you've done, it's a reasonable tradeoff.

Reply:
Thank you @elazzabi and @anu-rock ! I had tried the `sanitize_key()` function earlier and passed that since it was removing the spaces and affecting reliability. Thanks to the reminder here about excluding any other special characters though! 

I think the solution here would be to first replace the spaces with underscores, and then run that through `sanitize_key`. I will test this and place for further review here 👍

Reply:
> I think the solution here would be to first replace the spaces with underscores

We may skip this entirely. Why do you think readability is important here?

Reply:
> We may skip this entirely. Why do you think readability is important here?

My gut feel was if someone searches the db, they would look for `tax_with_spaces` instead of `taxwithspaces`. Coming to think of it if they look for `%spaces%` they will still find it. 

I missed reading your comment and for now I have passed the existing `$sanitized_type` formula through `sanitize_key()`.  Let me know if you would want me to keep it simple and move it to just `sanitize_key()` without bothering about underscores.